src/Uno.Toolkit.RuntimeTests/Tests/ItemsRepeaterExtensionTests.ISupportIncrementalLoading.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Toolkit.RuntimeTests.Helpers;
+using Uno.Toolkit.UI;
+using Uno.UI.RuntimeTests;
+using Windows.Foundation;
+using System.Collections.ObjectModel;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Controls;
+#else
+using ItemsRepeater = Microsoft.UI.Xaml.Controls.ItemsRepeater;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Controls;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests;
+
+partial class ItemsRepeaterExtensionTests
+{
+	[TestMethod]
+	[RequiresFullWindow]
+	[RunsOnUIThread]
+	[DataRow(Orientation.Vertical)]
+	[DataRow(Orientation.Horizontal)]
+	public async Task When_Incremental_Load(Orientation orientation)
+	{
+		const int BatchSize = 25;
+
+		var source = new InfiniteSource<int>(async start =>
+		{
+			await Task.Delay(25);
+			return Enumerable.Range(start, BatchSize).ToArray();
+		});
+
+		var SUT = new Grid() { Height = 210, Width = 210, VerticalAlignment = VerticalAlignment.Bottom };
```

Comment:
additionally, since your setup is likely to be rigid, you can convert them into a single method with this signature:
```cs
(Panel Setup, ItemsRepeater IR, ScrollViewer SV) BuildSetup(Object source, Orientation orientation)
```